rpcserver.go

```diff
@@ -6841,3 +7185,133 @@ func getDecimalDisplayNonStrict(
 
 	return fn.Some(decDisplay), nil
 }
+
+// rfqChannel returns the channel to use for RFQ operations. If a peer public
+// key is specified, the channels are filtered by that peer. If there are
+// multiple channels for the same asset, the user must specify the peer public
+// key.
+func (r *rpcServer) rfqChannel(ctx context.Context, id asset.ID,
+	peerPubKey *route.Vertex) (*channelWithAsset, error) {
+
+	balances, err := r.computeChannelAssetBalance(ctx)
+	if err != nil {
+		return nil, fmt.Errorf("error computing available asset "+
+			"channel balance: %w", err)
+	}
+
+	assetBalances, haveBalance := balances[id]
+	if !haveBalance || len(assetBalances) == 0 {
+		return nil, fmt.Errorf("no asset channel balance found for "+
+			"asset %s", id.String())
+	}
+
+	// If a peer public key was specified, we always want to use that to
+	// filter the asset channels.
+	if peerPubKey != nil {
+		assetBalances = fn.Filter(
+			assetBalances, func(c channelWithAsset) bool {
+				return c.channelInfo.PubKeyBytes == *peerPubKey
+			},
+		)
+	}
+
+	switch {
+	// If there are multiple asset channels for the same asset, we need to
+	// ask the user to specify the peer public key. Otherwise, we don't know
+	// who to ask for a quote.
```

Comment:
nit: we could convert this into a feature in next versions, the user may not want to pick a single peer here so instead we could ask for a quote from all peers and choose the "best" one

Reply:
I agree. But that would be a larger overhaul of the whole RFQ system, so I'd rather do that in a follow-up PR.